java-security/src/main/java/com/sap/cloud/security/token/validation/validators/XsuaaJwtIssuerValidator.java

```diff
@@ -38,27 +37,37 @@
 	@Override
 	public ValidationResult validate(Token token) {
 		String tokenKeyUrl = token.getHeaderParameterAsString(TokenHeader.JWKS_URL);
+		URI jkuUri;
+
 		if (tokenKeyUrl == null || tokenKeyUrl.trim().isEmpty()) {
 			return createInvalid(
 					"Issuer validation can not be performed because Jwt token does not contain 'jku' header parameter.");
 		}
-		return matchesTokenKeyUrlDomain(tokenKeyUrl);
-	}
-
-	private ValidationResult matchesTokenKeyUrlDomain(String tokenKeyUrl) {
-		URI jkuUri;
 		try {
-			jkuUri = new URI(tokenKeyUrl);
-			if (jkuUri.getHost() != null && jkuUri.getHost().endsWith(domain)) {
-				return createValid();
-			}
-		} catch (URISyntaxException e) {
-			logger.error("Error: 'jku' header parameter '{}' does not provide a valid URI: {}.", tokenKeyUrl,
-					e.getMessage(), e);
+			jkuUri = URI.create(tokenKeyUrl);
+		} catch (IllegalArgumentException e) {
+			return createInvalid(
+					"Issuer validation can not be performed because Jwt token does not contain a valid uri as 'jku' header parameter.");
+		}
+		if (!matchesTokenKeyUrlDomain(jkuUri)) {
+			return createInvalid(
+					"Issuer is not trusted because 'jku' '{}' does not match uaa domain '{}' of the identity service.",
+					jkuUri, domain);
+		}
+		if (!matchesTokenKeyEndpoint(jkuUri)) {
+			return createInvalid(
```

Comment:
only thing I could imagine is that a more detailed logging would be interessting. (e.g. because of URI parameters the jku does not match...)

Reply:
createInvalid() logs the message with all details.